File: admin/src/main/java/com/jigumulmi/place/dto/response/AdminPlaceBusinessHourResponseDto.java
package com.jigumulmi.place.dto.response;

import com.jigumulmi.place.dto.WeeklyBusinessHourDto;
import com.jigumulmi.place.dto.request.AdminCreateTemporaryBusinessHourRequestDto;
import java.util.List;
import lombok.Builder;
import lombok.Getter;
import lombok.experimental.SuperBuilder;

@Getter
@Builder
public class AdminPlaceBusinessHourResponseDto {

    @Getter
    @SuperBuilder
    public static class TemporaryBusinessHourDto extends AdminCreateTemporaryBusinessHourRequestDto {
        private Long id;
    }

    private WeeklyBusinessHourDto fixedBusinessHour;
    private List<TemporaryBusinessHourDto> temporaryBusinessHour;

    public static AdminPlaceBusinessHourResponseDto from(WeeklyBusinessHourDto fixedBusinessHour,
        List<TemporaryBusinessHourDto> temporaryBusinessHourList) {
        return AdminPlaceBusinessHourResponseDto.builder()
            .fixedBusinessHour(fixedBusinessHour)
            .temporaryBusinessHour(temporaryBusinessHourList)
            .build();
    }

}

File: admin/src/main/kotlin/com/jigumulmi/place/dto/response/AdminPlaceBusinessHourResponseDto.kt
package com.jigumulmi.place.dto.response

import com.jigumulmi.place.dto.BusinessHour
import com.jigumulmi.place.dto.WeeklyBusinessHourDto
import com.jigumulmi.place.dto.validator.ValidBusinessHour
import jakarta.validation.constraints.FutureOrPresent
import jakarta.validation.constraints.NotNull
import java.time.LocalDate

data class AdminPlaceBusinessHourResponseDto (
    val fixedBusinessHour: WeeklyBusinessHourDto,
    val temporaryBusinessHour: List<TemporaryBusinessHourDto>
) {

    data class TemporaryBusinessHourDto (
        val id: Long,
        val date: @NotNull @FutureOrPresent LocalDate,

        @ValidBusinessHour
        val businessHour: BusinessHour,
    )

    companion object {
        fun from(
            fixedBusinessHour: WeeklyBusinessHourDto,
            temporaryBusinessHourList: List<TemporaryBusinessHourDto>
        ): AdminPlaceBusinessHourResponseDto {
            return AdminPlaceBusinessHourResponseDto(
                fixedBusinessHour = fixedBusinessHour,
                temporaryBusinessHour = temporaryBusinessHourList
            )
        }
    }
}
